packages/nimble-components/src/table-column/number-text/models/unit-scale-formatter.ts

```diff
@@ -0,0 +1,96 @@
+import type { ScaledUnit } from './scaled-unit';
+import { NumberFormatter } from './number-formatter';
+
+export type UnitScaleFormatterContructor = new (
+    lang: string,
+    formatterOptions: Intl.NumberFormatOptions
+) => UnitScaleFormatter;
+
+/**
+ * A formatter for a number-text column that supports unit labels.
+ */
+export abstract class UnitScaleFormatter extends NumberFormatter {
+    public alwaysUseBaseUnit = false;
+
+    private get supportedUnits(): ScaledUnit[] {
+        const values = this.setSupportedUnitsAndBaseUnit();
+        return values.supportedUnits;
+    }
+
+    private get baseUnit(): ScaledUnit {
+        const values = this.setSupportedUnitsAndBaseUnit();
+        return values.baseUnit;
+    }
+
+    public constructor(
+        private readonly lang: string,
+        private readonly formatterOptions: Intl.NumberFormatOptions
+    ) {
+        super();
+    }
+
+    public getValueForBestUnit(number: number): number {
+        const unit = this.pickBestUnit(number);
+        return number / unit.conversionFactor;
+    }
+
+    protected abstract getSupportedUnits(
+        lang: string,
+        formatterOptions: Intl.NumberFormatOptions
+    ): ScaledUnit[];
+
+    protected override format(number: number): string {
+        const unit = this.pickBestUnit(number);
+        return unit.format(number / unit.conversionFactor);
+    }
+
+    private pickBestUnit(number: number): ScaledUnit {
```

Comment:
It looks like every number that gets formatted calls this function twice -- once when calling `getValueForBestUnit` and once when calling `format`. Instead, is there a way to rewrite part of this code so that doesn't have to happen. In most cases, it seems pretty straightforward, but the `alwaysUseBaseUnit` code path might make things too complicated.

Reply:
> is there a way to rewrite part of this code so that doesn't have to happen. In most cases, it seems pretty straightforward...

I may be dense, but I'm not seeing a straightforward way. I really dislike that I had to add `getValueForBestUnit()` to the API and run `pickBestUnit()` twice, but it seems necessary. The only thing I can think of is to keep a cache of calculated best units for given input numbers, but that is too much. Let me know if you had something else in mind.

Reply:
The case in the `decimal` formatter seems fairly straightforward, but I agree that the `default` formatter definitely makes things more difficult.

I don't have a huge issue with the code as-is as long as it isn't causing performance problems. It just seems like there is a lot of work being done in order to render a single cell value.